File: Fresh/Fresh.Services/Database/Payment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Fresh.Services.Database
{
    public class Payment
    {
        [Key]
        public int Id { get; set; }
        public decimal Amount { get; set; }
        public DateTime PaymentDate { get; set; }
        public bool IsFinalPayment { get; set; }

        [ForeignKey(nameof(PurchaseId))]
        public Purchase Purchase { get; set; }
        public int PurchaseId { get; set; }
    }
}

File: Fresh/UI/fresh_app/src/app/model/Payment.ts
export class Payment {
    id: number;
    amount: number;
    paymentDate: Date;
    isFinalPayment: boolean;
    purchaseId: number;

    constructor(id: number,
        amount: number,
        paymentDate: Date,
        isFinalPayment: boolean,
        purchaseId: number) {
        this.id = id;
        this.amount = amount;
        this.paymentDate = paymentDate;
        this.isFinalPayment = isFinalPayment;
        this.purchaseId = purchaseId;
    }
}